common/lib/dependabot/config/ignore_condition.rb

```diff
@@ -57,54 +57,20 @@ def versions_by_type(dependency)
         version = correct_version_for(dependency)
         return [] unless version
 
-        semver = version.to_semver
-
         transformed_update_types.flat_map do |t|
           case t
           when PATCH_VERSION_TYPE
-            ignore_patch(semver)
+            version.ignored_patch_versions
```

Comment:
I'm proposing to move the code that calculates the range to be ignored into version because in most cases it will be ecosystem dependent.

Reply:
Makes sense; this is a bigger change to some underlying assumptions and a tricky area; can you get more input from folks on the team? 

Tagging @jakecoffman and @JamieMagee for awareness